pkg/network/encoding/encoding_test.go

```diff
@@ -365,6 +369,9 @@ func testSerialization(t *testing.T, aggregateByStatusCode bool) {
 		sort.Strings(result.Tags)
 		// fixup: json marshaler encode nil slice as empty
 		result.Conns[0].Tags = nil
+		// fixup: json marshaler with encode nil map as empty
+		result.Conns[0].TcpFailuresByErrCode = nil
```

Comment:
I think this might help? https://github.com/DataDog/datadog-agent/blob/main/pkg/network/encoding/unmarshal/json.go#L58-L70

Reply:
Yes, definitely